core/components/menu/menu.js

```diff
@@ -352,52 +262,43 @@ Blockly.Menu.prototype.highlightNext = function() {
  * @package
  */
 Blockly.Menu.prototype.highlightPrevious = function() {
-  this.unhighlightCurrent();
-  this.highlightHelper(function(index, max) {
-    index--;
-    return index < 0 ? max - 1 : index;
-  }, this.highlightedIndex_);
+  var index = this.menuItems_.indexOf(this.highlightedItem_);
+  this.highlightHelper(index < 0 ? this.menuItems_.length : index, -1);
+};
+
+/**
+ * Highlights the first highlightable item.
+ * @package
+ */
+Blockly.Menu.prototype.highlightFirst = function() {
+  this.highlightHelper(-1, 1);
+};
+
+/**
+ * Highlights the last highlightable item.
+ * @package
+ */
+Blockly.Menu.prototype.highlightLast = function() {
+  this.highlightHelper(this.menuItems_.length, -1);
 };
 
 /**
  * Helper function that manages the details of moving the highlight among
  * child menuitems in response to keyboard events.
- * @param {function(this: Blockly.Component, number, number) : number} fn
- *     Function that accepts the current and maximum indices, and returns the
- *     next index to check.
  * @param {number} startIndex Start index.
- * @return {boolean} Whether the highlight has changed.
+ * @param {number} delta Step direction: 1 to go down, -1 to go up.
  * @protected
  */
-Blockly.Menu.prototype.highlightHelper = function(fn, startIndex) {
-  // If the start index is -1 (meaning there's nothing currently highlighted),
-  // try starting from the currently open item, if any.
-  var curIndex =
-      startIndex < 0 ? -1 : startIndex;
-  var numItems = this.getChildCount();
-
-  curIndex = fn.call(this, curIndex, numItems);
-  var visited = 0;
-  while (visited <= numItems) {
-    var menuItem = /** @type {Blockly.MenuItem} */ (this.getChildAt(curIndex));
-    if (menuItem && this.canHighlightItem(menuItem)) {
-      this.setHighlightedIndex(curIndex);
-      return true;
+Blockly.Menu.prototype.highlightHelper = function(startIndex, delta) {
```

Comment:
Since this method is @protected, it should have an underscore at the end.

Reply:
Ok, this is new.  None of the current @protected methods in core/components have an underscore.  I'll change them all.

Reply:
Yes, new as in it wasn't done because core/components mostly was inherited from closure, but course not new to Blockly (as the standard).